tests/mlmodel_sklearn/test_cluster_models.py

```diff
@@ -0,0 +1,186 @@
+        "SpectralCoclustering",
+        "SpectralClustering",
+    ],
+)
+def test_below_v1_1_model_methods_wrapped_in_function_trace(cluster_model_name, run_cluster_model):
+    expected_scoped_metrics = {
+        "AffinityPropagation": [
+            ("Function/MLModel/Sklearn/Named/AffinityPropagation.fit", 2),
+            ("Function/MLModel/Sklearn/Named/AffinityPropagation.predict", 1),
+            ("Function/MLModel/Sklearn/Named/AffinityPropagation.fit_predict", 1),
+        ],
+        "AgglomerativeClustering": [
+            ("Function/MLModel/Sklearn/Named/AgglomerativeClustering.fit", 2),
+            ("Function/MLModel/Sklearn/Named/AgglomerativeClustering.fit_predict", 1),
+        ],
+        "Birch": [
+            ("Function/MLModel/Sklearn/Named/Birch.fit", 2),
+            (
+                "Function/MLModel/Sklearn/Named/Birch.predict",
+                1 if SKLEARN_VERSION >= (1, 0, 0) else 3,
+            ),
+            ("Function/MLModel/Sklearn/Named/Birch.fit_predict", 1),
+            ("Function/MLModel/Sklearn/Named/Birch.transform", 1),
+        ],
+        "DBSCAN": [
+            ("Function/MLModel/Sklearn/Named/DBSCAN.fit", 2),
+            ("Function/MLModel/Sklearn/Named/DBSCAN.fit_predict", 1),
+        ],
+        "FeatureAgglomeration": [
+            ("Function/MLModel/Sklearn/Named/FeatureAgglomeration.fit", 1),
+            ("Function/MLModel/Sklearn/Named/FeatureAgglomeration.transform", 1),
+        ],
+        "KMeans": [
+            ("Function/MLModel/Sklearn/Named/KMeans.fit", 2),
+            ("Function/MLModel/Sklearn/Named/KMeans.predict", 1),
+            ("Function/MLModel/Sklearn/Named/KMeans.fit_predict", 1),
+            ("Function/MLModel/Sklearn/Named/KMeans.transform", 1),
+        ],
+        "MeanShift": [
+            ("Function/MLModel/Sklearn/Named/MeanShift.fit", 2),
+            ("Function/MLModel/Sklearn/Named/MeanShift.predict", 1),
+            ("Function/MLModel/Sklearn/Named/MeanShift.fit_predict", 1),
+        ],
+        "MiniBatchKMeans": [
+            ("Function/MLModel/Sklearn/Named/MiniBatchKMeans.fit", 2),
+            ("Function/MLModel/Sklearn/Named/MiniBatchKMeans.predict", 1),
+            ("Function/MLModel/Sklearn/Named/MiniBatchKMeans.fit_predict", 1),
+        ],
+        "SpectralBiclustering": [
+            ("Function/MLModel/Sklearn/Named/SpectralBiclustering.fit", 1),
+        ],
+        "SpectralCoclustering": [
+            ("Function/MLModel/Sklearn/Named/SpectralCoclustering.fit", 1),
+        ],
+        "SpectralClustering": [
+            ("Function/MLModel/Sklearn/Named/SpectralClustering.fit", 2),
+            ("Function/MLModel/Sklearn/Named/SpectralClustering.fit_predict", 1),
+        ],
+    }
+    expected_transaction_name = "test_cluster_models:_test"
+    if six.PY3:
+        expected_transaction_name = (
+            "test_cluster_models:test_below_v1_1_model_methods_wrapped_in_function_trace.<locals>._test"
```

Comment:
We need to add those two other models into here: Ward and WardAgglomeration. I'm not sure exactly what version those got removed. I saw them in version 0.15.0.

Reply:
I believe the support window for this is 3 years, so anything before 2019 would not be supported anyway (seems like v0.20 is a good stopping point for us now).  @ak-war can you confirm?